File: topics/type-checking/examples.js
#!/usr/bin/env node

console.log("PYTHAGORAS");

function pythagoras(a, b) {
    console.log(Math.sqrt((a * a) + (b * b)));
}

pythagoras(2, 3);
pythagoras("2", 3);
pythagoras(true + true, 3);
pythagoras("shoe", 3);
pythagoras(null, 3);

console.log("\nREPEAT");

function repeat(msg, count) {
    var output = null;
    for (i = 0; i < count; i++) {
        if (output === null) {
            output = "";
        } else {
            output += " ";
        }
        output += msg;
    }
    console.log(output);
}

repeat("hello", 3);
repeat("hello", "3");
repeat("hello", true + true + true);
repeat("hello", "duck");
repeat(null, 3);

File: topics/data-types/examples.py
#!/usr/bin/env python3

print("PYTHAGORAS")

def pythagoras(a: int, b: int) -> None:
    from math import sqrt
    print(sqrt((a * a) + (b * b)))

pythagoras(2, 3)
pythagoras("2", 3)
pythagoras(True + True, 3)
pythagoras("shoe", 3)
pythagoras(None, 3)

print("\nREPEAT")

def repeat(msg: str, count: int) -> None:
    output = None
    for i in range(count):
        if output is None:
            output = ""
        else:
            output += " "
        output += msg
    print(output)

repeat("hello", 3)
repeat("hello", "3")
repeat("hello", True + True + True)
repeat("hello", "duck")
repeat(None, 3)
